Patch 3

strider820 opened the pull request.

Commit:
Proposed changes/fixes

- make the renderer customizable via options passed in a config array (see explanation in the doc.md)
- I am forcing the parameters slot, content_width and content_height to be mandatory when outstream is used
- I have added some checks in the rendering function to make sure that slot and content_width/content_height are properly used.

Commit:
All EASI player parameters can be used now

- I have added an parameter called custom_override which let you custom all easi params you want

strider820 commented on bin/spotxBidAdapter.js:
```diff
@@ -49,8 +49,8 @@ export const spec = {
         }
-        if (!utils.getBidIdParameter('video_slot', bid.params.outstream_options) || !utils.getBidIdParameter('content_width', bid.params.outstream_options) || !utils.getBidIdParameter('content_height', bid.params.outstream_options)) {
-          utils.logError(BIDDER_CODE + ': please define parameters video_slot, content_width and content_height for object outstream_options in the configuration.');
+        if (!(utils.getBidIdParameter('slot', bid.params.outstream_options) && utils.getBidIdParameter('content_width', bid.params.outstream_options) && utils.getBidIdParameter('content_height', bid.params.outstream_options))) {
```
the first && needs to be ||, and the last two conditions need to be in parentheses

jmonbillard replied:
I think the 3 options have to be mandatory for the outstream rendering. Don't you think?

For the adcall checks, I made channel_id and playerSize mandatory whereas for outstream unit option I made also slot/width/height mandatory for displaying the renderer.

strider820 replied:
nope, just one or the other. Easi can figure out the width and height if all we have is a content_container

strider820 commented on bin/spotxBidAdapter.js:
```diff
@@ -292,27 +292,63 @@ function outstreamRender(bid) {
+        script.setAttribute('data-spotx_collapse', '0');
+        script.setAttribute('data-spotx_autoplay', '1');
+        script.setAttribute('data-spotx_blocked_autoplay_override_mode', '1');
+        script.setAttribute('data-spotx_video_slot_can_autoplay', '1');
```
add all these defaults to if they have customOverride defined. So that, if they only want to override autoplay, they don't also have to override collapse to keep the previous behavior

strider820 commented on bin/spotxBidAdapter.md:
```diff
@@ -30,7 +30,47 @@ This adapter requires setup and approval from the SpotX team.
+                    custom_override: [{ // This option is not mandatory though used to override default renderer parameters using EASI player options in here: https://developer.spotxchange.com/content/local/docs/sdkDocs/EASI/README.md
+                        name: 'content_width',
+                        value: 640
+                    }, {
```
let's pare this down to just an example or two, rather than spelling out all of them

strider820 commented on bin/spotxBidAdapter.js:
```diff
@@ -49,8 +49,8 @@ export const spec = {
-        if (!utils.getBidIdParameter('video_slot', bid.params.outstream_options) || !utils.getBidIdParameter('content_width', bid.params.outstream_options) || !utils.getBidIdParameter('content_height', bid.params.outstream_options)) {
-          utils.logError(BIDDER_CODE + ': please define parameters video_slot, content_width and content_height for object outstream_options in the configuration.');
+        if (!(utils.getBidIdParameter('slot', bid.params.outstream_options) && utils.getBidIdParameter('content_width', bid.params.outstream_options) && utils.getBidIdParameter('content_height', bid.params.outstream_options))) {
+          utils.logError(BIDDER_CODE + ': please define parameters slot, content_width and content_height for outstream_options object in the configuration.');
```
"please define parameters slot or content_width and content_height"

jmonbillard replied:
same as above comment. I think these 3 should be mandatory all together for displaying purposes. Let me know your thoughts

Commit:
new fixes and changes

- now I am "really" overriding the customOverride parameters to default ones
- maybe we should remove some default parameters then?

Commit:
Update spotxBidAdapter.md

Commit:
Update spotxBidAdapter.md

Commit:
removed contetn_width/height from mandatory fields

Commit:
change the way I was pushing elements in dataSpotXParams
